src/App.js

```diff
@@ -48,21 +77,53 @@ const App = () => {
     // 3. Go across each diagonal to see if 
     //    all three squares have the same value.
 
+
+    for(let i = 0; i < squares.length; i ++){
+      if (squares[i][0].value === squares[i][1].value && squares[i][1].value === squares[i][2].value && squares[i][0].value !== ''){
+        setWinner(squares[i][0].value)
+      }
+      else if (squares[0][i].value === squares[1][i].value && squares[1][i].value === squares[2][i].value && squares[1][i].value !==''){
+        setWinner(squares[0][i].value)
+      }
+    }
+    // diaognal
+    if(squares[0][0].value === squares[1][1].value && squares[1][1].value === squares[2][2].value && squares[1][1] !== ''){
+      setWinner(squares[0][0].value)
+    } else if(squares[2][0].value === squares[1][1].value && squares[1][1].value === squares[0][2].value && squares[1][1] !== ''){
+      setWinner(squares[2][0].value)
+    }
+
+  }
```

Comment:
Nice job with this loop!  Getting it right is tricky.  :smiley:

Reply:
Thank you Kaida